socs/agent/smurf_recorder.py

```diff
@@ -21,6 +23,19 @@ class FlowControl(Enum):
     CLEANSE = 3
 
 
+def check_port(host, port, timeout=10):
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.settimeout(timeout)
+    s.setblocking(1)
```

Comment:
If I'm reading this right, isn't `s.setblocking(1)` equivalent to `s.settimeout(None)`. It seems like then the line above isn't really doing anything.

Reply:
Yeah, looking a bit more into and I am starting to think that I should just be using `s.setblocking(False)`, which would return immediately if the connect attempt fails. Like you say in another comment, the `_establish_reader_connection` function will manually prevent rapid connection attempts.

Reply:
After testing a bit, it seems like when the port is closed it sends `ConnectionRefusedError` immediately regardless of the timeout and blocking. My guess is that timeout and blocking are mostly for reading and writing to the port, and the connection is handled differently by the system. I'll use `blocking=True`, and `timeout=0.`